parlai/agents/seq2seq/seq2seq.py

```diff
@@ -318,12 +318,15 @@ def observe(self, observation):
         # shallow copy observation (deep copy can be expensive)
         obs = observation.copy()
         batch_idx = self.opt.get('batchindex', 0)
-        obs['text2vec'] = maintain_dialog_history(
-            self.history, obs,
-            reply=self.answers[batch_idx],
-            historyLength=self.opt['history_length'],
-            useReplies=self.opt['history_replies'],
-            dict=self.dict)
+        if not obs.get('preprocessed', False):
+            obs['text2vec'] = maintain_dialog_history(
+                self.history, obs,
+                reply=self.answers[batch_idx],
+                historyLength=self.opt['history_length'],
+                useReplies=self.opt['history_replies'],
+                dict=self.dict)
+        else:
+            obs['text2vec'] = deque(obs['text2vec'], self.opt['history_length'])
```

Comment:
wouldn't that already be part of the preprocessing?

Reply:
i guess it depends how much preprocessing is done.. there are levels...?

Reply:
A deque is not serializable for saving JSON, so when I save the data it needs to be converted to a list; this converts it back to a deque so it can be used accordingly later down the model's pipeline